components/GetHistory/GetHistory.tsx

```diff
@@ -0,0 +1,156 @@
+        const newState :any = {}
+        newState.cells = TicTacToeBoard.fromEncoded(argument.signedMove.gameMove.newState).cells
+        newState.crossesWin = TicTacToeBoard.fromEncoded(argument.signedMove.gameMove.newState).crossesWin
+        newState.naughtsWin = TicTacToeBoard.fromEncoded(argument.signedMove.gameMove.newState).naughtsWin
+        oldState.cells = arrayToString(toInt(oldState.cells))
+        newState.cells = arrayToString(toInt(newState.cells))
+        const move = defaultAbiCoder.decode(TIC_TAC_TOE_MOVE_TYPES, argument.signedMove.gameMove.move)
+        const formattedMove = formatMoves(move)
+  
+        const nonce = message.nonce
+        const polygonMessageGameMove = message.polygonMessageGameMove
+        const polygonMessageSignedMove = message.polygonMessageSignedMove
+        const signatureInfo = message.signatureInfo
+        return {newState, oldState, formattedMove, nonce, polygonMessageGameMove, polygonMessageSignedMove, signatureInfo}
+      }
+
+     if (gameType === 'checkers') {
+      const argument = message.message.arguments[0]
+      const oldState :any = {}
+      oldState.cells = CheckersBoard.fromEncoded(argument.signedMove.gameMove.oldState).cells
+      oldState.winner = CheckersBoard.fromEncoded(argument.signedMove.gameMove.oldState).winner
+      oldState.redMoves = CheckersBoard.fromEncoded(argument.signedMove.gameMove.oldState).redMoves
+      const newState :any = {}
+      newState.cells = CheckersBoard.fromEncoded(argument.signedMove.gameMove.newState).cells
+      newState.winner = CheckersBoard.fromEncoded(argument.signedMove.gameMove.newState).winner
+      newState.redMoves = CheckersBoard.fromEncoded(argument.signedMove.gameMove.newState).redMoves
+      oldState.cells = arrayToString(toInt(oldState.cells))
+      newState.cells = arrayToString(toInt(newState.cells))
+      const move = defaultAbiCoder.decode(CHECKERS_MOVE_TYPES, argument.signedMove.gameMove.move)
+      const formattedMove = formatMoves(move)
+
+      const nonce = message.nonce
+      const polygonMessageGameMove = message.polygonMessageGameMove
+      const polygonMessageSignedMove = message.polygonMessageSignedMove
+      const signatureInfo = message.signatureInfo
+      return {newState, oldState, formattedMove, nonce, polygonMessageGameMove, polygonMessageSignedMove, signatureInfo}
+    }
+    })
+    
+    const historyData = {
+      gameId,
+      gameType,
+      messages,
+      historyMessages,
+    }
+
+    const currentDate = new Date();
+    const formattedDate = currentDate.getDate() + "." + (currentDate.getMonth() + 1) + "." + currentDate.getFullYear()
+     + "_" + currentDate.getHours() + "." + currentDate.getMinutes();
+
+    const historyDataYaml = YAML.stringify(historyData)
+
+    const a = document.createElement("a")
+    const file = new Blob([historyDataYaml], {type: 'text/plain'});
+    a.href = URL.createObjectURL(file);
+    a.download = `gamerecord_${formattedDate}.txt`;
```

Comment:
Which game, checkers or ttt? Who played it?
Please make it 
gj_arbiter_0x1234_checkers_0x1234_<game id>_<date>_<0xabcd>_<0x1234>. yaml

Reply:
ok

Reply:
changed to "gamerecord_0x1f0b_checkers_731_0xdC5f_0x4114_2023-02-03_15-24.yaml"